File: geral/entendendoalgoritmos/chapter_001/binarysearch.rb
def binary_search(values, target)
  left = 0

  right = values.length - 1

  while left <= right
    mid = (left + right) / 2

    if values[mid] == target
      return true
    elsif values[mid] > target
      right = mid - 1
    else
      left = mid + 1
    end
  end

  return false
end

values = [1, 3, 5, 7, 9]

puts binary_search(values, 7)

File: geral/livro_entendendo_algoritmos/chapter_001/binarysearch.go
package main

import "fmt"

func main() {
	values := []int{1, 3, 5, 7, 9}

	fmt.Println(binarySearch(values, 7))
}

func binarySearch(values []int, target int) bool {
	left := 0

	right := len(values) - 1

	for left <= right {
		mid := (left + right) / 2

		if values[mid] == target {
			return true
		} else if values[mid] > target {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}

	return false
}
